rewrite-core/src/main/java/org/openrewrite/internal/RemoveAfter.java

```diff
@@ -0,0 +1,32 @@
+/*
+ * Copyright 2025 the original author or authors.
+ * <p>
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * <p>
+ * https://www.apache.org/licenses/LICENSE-2.0
+ * <p>
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.openrewrite.internal;
+
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+
+/**
+ * An annotation that indicates that the element should be removed after a certain point.
+ * Intended to be used with a matching recipe to clear out elements on a schedule.
+ */
+@Retention(RetentionPolicy.RUNTIME)
```

Comment:
Since we only need recipes to operate on this, we might as well use SOURCE retention, right?

Reply:
Also we should allow the annotation to target anything, probably.

Reply:
My thought of having it be runtime would be that we can also evaluate and lock down at runtime if we have such a need; something of a kill switch.

Reply:
I think that if we in a first step just can get the best practices recipe or a similar mechanism to report that we have some code that can be removed, that would already be a very good start.